### DISCUSSION TITLE
Should we check if `startTime` is zero in LD and LT?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#851] [General]
In `LockupLinear`, we no longer allow `startTime == 0`. This change has been implemented because we now set `cliffTime == 0` to create a [linear curve stream](https://docs.sablier.com/concepts/protocol/stream-types#lockup-linear).

However, this has introduced inconsistency across contracts. In `LockupDynamic` and `LockupTranched`, the start time can still be zero.

Should we also disallow a start time of zero in `LockupDynamic` and `LockupTranched`? 
What are your thoughts? @PaulRBerg @smol-ninja

**Comment by PaulRBerg** [comment]
> Should we also disallow a start time of zero in `LockupDynamic` and `LockupTranched`?

Yes

---

Separately - I think you should have tagged @sablier-labs/engineers since this discussion is relevant for the front-end; cc @razgraf, @gavriliumircea.

**Reply by andreivladbrg** [reply]
>  I think you should have tagged @sablier-labs/engineers since this discussion is relevant for the front-end; cc @razgraf, @gavriliumircea.

yeah, my bad

**Reply by razgraf** [reply]
I don't think it affects us in any way (the UI doesn't allow start times of 0 anyway), so for the sake of consistency, go for it! 

P.S. However, please do include this in the changelog we should follow for the v2.1 to v2.2 upgrade

**Comment by andreivladbrg** [comment]
Ended up implementing this in https://github.com/sablier-labs/v2-core/pull/852
